### PR TITLE
Osman Elsahib week 7 - Homework

### PR DESCRIPTION
**Pull request by elsahib**
# Your Details

Your Name: Osman Elsahib
Your City: Manchester
Your Slack Name: Osman Elsahib

# Homework Details

Module: Js 
Week: 7

### CODE REVIEW

File: week-7/Homework/mandatory/0-thinking-like-a-programmer/task.md [hunk 1]
```diff
@@ -27,11 +27,12 @@ After you've watched these videos I'd like you to answer these questions
...
+Problem Solving
 
 ## 2. When trying to solve a challenge, what should you do first?
-
+try to process the problem bit by bit, take small part and solve it then take another part and so on untill you finish
 <!-- Write your answer here -->
 
 ## 3. What should you do if you get stuck?
-
+seek help, look online, ask a peer or a mentor. If non could help refer to documentation.
```
Comment (zoltan-gal) [inline week-7/Homework/mandatory/0-thinking-like-a-programmer/task.md:37]: Yep, documentation is a good point especially when we need specific details. Although it is tedious to read sometimes, it should be the source of truth.

File: week-7/Homework/mandatory/1-debugging-practice/script.js [hunk 2]
```diff
@@ -37,8 +37,8 @@ function submit() {
     alert("Please fill all fields!");
```
Comment (zoltan-gal) [inline week-7/Homework/mandatory/1-debugging-practice/script.js:37]: Only one of the acceptance criteria has been missed out
> If any of the information is missing it shouldn't add the book and should show an alert
 In the if statement author needs to be checked too:
```
  if (
    title.value == null ||
    title.value == "" ||
    author.value == null ||
    author.value == "" ||
    pages.value == null ||
    pages.value == ""
  ) {
```

File: week-7/Homework/mandatory/0-thinking-like-a-programmer/task.md (hunk 1 above)
Comment (zoltan-gal) [reply week-7/Homework/mandatory/0-thinking-like-a-programmer/task.md:37]: Seek for help in the right time and not getting stuck too long in the rabbit hole.

File: week-7/Homework/mandatory/1-debugging-practice/script.js (hunk 2 above)
Comment (elsahib) [reply week-7/Homework/mandatory/1-debugging-practice/script.js:37]: I didn't see this to be honest, I thought I should only clear the console from errors

**Pull request closed without merging**